core/block_validator.go

```diff
@@ -41,15 +41,17 @@ var (
 //
 // BlockValidator implements Validator.
 type BlockValidator struct {
-	bc  *BlockChain // Canonical block chain
-	Pow pow.PoW     // Proof of work used for validating
+	config *ChainConfig
+	bc     *BlockChain // Canonical block chain
+	Pow    pow.PoW     // Proof of work used for validating
 }
 
 // NewBlockValidator returns a new block validator which is safe for re-use
-func NewBlockValidator(blockchain *BlockChain, pow pow.PoW) *BlockValidator {
+func NewBlockValidator(config *ChainConfig, blockchain *BlockChain, pow pow.PoW) *BlockValidator {
```

Comment:
As discussed, the config field for this method can be retrieved from the blockchain, so duplicating the data would just open up the potential for strange programming errors when the two mismatch. Please from config and fetch it from blockchain.